UnitsNet/Serialization/Contracts/ExtendedQuantityValueContract.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed under MIT No Attribution, see LICENSE file at the root.
+// Copyright 2013 Andreas Gullberg Larsen (andreas.larsen84@gmail.com). Maintained at https://github.com/angularsen/UnitsNet.
+
+using System.Runtime.Serialization;
+
+namespace UnitsNet.Serialization.Contracts
+{
+    /// <summary>
+    /// An extension to the basic quantity contract that allows for dual representation of a non-standard quantity (e.g. <see cref="IDecimalQuantity"/>).
+    /// This contract is provided for compatibility with the JsonNet representation (see https://github.com/angularsen/UnitsNet#serialization)
+    /// </summary>
+    [DataContract]
+    public class ExtendedQuantityValueContract<TValueType, TUnit>: QuantityValueContract<double, TUnit>
+    {
+        /// <summary>
+        /// Initializes a new quantity with the given standard unit and value pair, plus their extended representation
+        /// </summary>
+        /// <param name="value">The standard (double) value</param>
+        /// <param name="unit">The associated unit</param>
+        /// <param name="valueString">The string representation of the actual value type</param>
+        /// <param name="valueType">The value type associated with this given string representation</param>
+        public ExtendedQuantityValueContract(double value, TUnit unit, string valueString, TValueType valueType) : base(value, unit)
+        {
+            ValueString = valueString;
+            ValueType = valueType;
+        }
+
+        /// <summary>
+        /// The string representation of the actual value type.
+        /// </summary>
+        [DataMember(Order = 3)]
+        public string ValueString { get; protected set; }
+
+        /// <summary>
+        /// The value type associated with this given string representation.
+        /// </summary>
+        [DataMember(Order = 4)]
+        public TValueType ValueType { get; protected set; }
+    }
```

Comment:
Codecov points out some of these data contract members may be lacking test coverage. If that is correct, I think we should add some basic serialization tests for these.

Reply:
I tried a few ways of "covering" those. My final attempt has both the [QuantityValueContract_Constructor_AssignsAllProperties](https://github.com/lipchev/UnitsNet/blob/a7e0bc60f9d70b0d320f88f11b8fe868972e1158/UnitsNet.Serialization.DataContract.Tests/QuantityValueDataContractTests.cs#L12) and [QuantityValueContract_SerializationRoundTrips](https://github.com/lipchev/UnitsNet/blob/a7e0bc60f9d70b0d320f88f11b8fe868972e1158/UnitsNet.Serialization.DataContract.Tests/DataContractSerializerTests/DataContractSerializerTests.cs#L80) (one for each DC serializer).
However now that I've also pushed the generated quantities - the code-coverage fails because of the merge conflicts.. :(